app/helpers/candidates_helper.rb

```diff
@@ -17,102 +17,143 @@ def candidate_display_activity(event)
     actor_account_name = compose_actor_account_name(event)
 
     text =
-      if event.type == "email_received"
-        []
-      else
-        [actor_account_name]
-      end
```

Comment:
This code also started to branch more and more. I don't like it.
I tried changing it. But without tests it looked dangerous. So I added tests too.